### ISSUE TITLE
Guide For Beginners Very Important - Send Socket before connecting

### ISSUE THREAD

**Issue by freeost** [n1nj4z33/iqoptionapi#52]
GUys please use the below code for sending socket and you will get your balance info, candlessss
THanks to me of course hahaha

Please get your cookie from google chrome . filter iqoption then look for ssid , copy the content and replace below

import urllib3
import time
from iqoptionapi.api import IQOptionAPI
from websocket import create_connection
import json
ws = create_connection("wss://iqoption.com/echo/{randomnumber}/{randomnumber}/websocket")
print "Sending 'Cookie'..."
ws.send("{'msg': 'MY COOKIE ID', 'name': 'ssid'}")
print "Sent"
print "Reeiving..."
result = ws.recv()
print "Received '%s'" % result

urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
api = IQOptionAPI("iqoption.com", "email_address", "Password")
api.connect()
time.sleep(5)
balance = api.profile.balance
time.sleep(5)
print balance
api.setactives([1, 2])
time.sleep(5)
api.getcandles(1,1)
time.sleep(5)
data = api.candles.candles_data
print api.candles.candles_data

--------------------------------------------
Output should be
------------------------------------------------

Sending 'Cookie...
Sent
Reeiving...
Received 'o'
30949.73
[[1506790640, 1181630, 1181630, 1181630, 1181630], [1506790641, 1181630, 1181630, 1181630, 1181630], [1506790642, 1181630, 1181630, 1181630, 1181630], [1506790643, 1181630, 0, 1181630, 1181630]]

**Comment by lucianopinheiro** [comment]
Still not working for me. I got the "Received 'o'" line, but no candles.
I am still able to do orders and get newChartData, but no candles.
:(

Should we replace "{randonnumber}" for an integer number (wich is the range) or let it literally this way?

**Comment by lucianopinheiro** [comment]
Nope. It is working again. Made some changes - not related with your solution.

**Comment by freeost** [comment]
@lucianopinheiro  Since something is wrong with sockets and I tried even with python request lib and sent exact same headers , still nothing works. I think they have done too much security. But anyway, I developed my own robot with C# now https://www.youtube.com/watch?v=aqjqVWU3XSk

**Comment by lucianopinheiro** [comment]
For me it is working again, thanks to your code. I just made a small change. Now, you should not ask for candles inside a list (sorry, I don't remember if it was part of original api).
Before: `api.getcandles([active.id])`
Now: `api.getcandles(active.id)`

This is my ws message sent
```
{ "msg": 
     {"duration": 60,
       "till": 1507844461,
       "chunk_size": 25,
       "from": 1507844340,
       "active_id": 1
      },
    "name": "candles"
}
```

The program is running right now:

```$ python main.py 
2017-10-12 18:22:07:INFO: Create connection.
ssid 'o'
2017-10-12 18:22:13:INFO: Successfully connected.
2017-10-12 18:22:13:INFO: Getting candles from 2017-10-12 17:52:00 till now.
2017-10-12 18:22:13:INFO: Now entering live mode.
2017-10-12 18:23:03:INFO: Inserted 12/12 candles: AUDJPY 3/3, EURUSD 3/3, USDJPY 3/3, AUDUSD 3/3 and 5 quotations
2017-10-12 18:24:02:INFO: Inserted 4/12 candles: AUDJPY 1/3, EURUSD 1/3, USDJPY 1/3, AUDUSD 1/3 and 12 quotations
2017-10-12 18:25:02:INFO: Inserted 4/12 candles: AUDJPY 1/3, EURUSD 1/3, USDJPY 1/3, AUDUSD 1/3 and 11 quotations
```

**Comment by zermeno98** [comment]
can´t get any candle  I receive the balance and also print the datetime form server but only a "NONE" msg for the CANDLES  can you help me I already tried the solution with the COOKIE info

thanks

**Comment by mateusmachadoinvestimentos** [comment]
OI @lucianopinheiro você pode me chamar no whatsapp ? estou criando uma API para c#, queria perguntar algumas coisas para você

**Comment by lucianopinheiro** [comment]
Manda. Não sei como comunicar diretamente por aqui.

**Comment by alexandreamigo01** [comment]
Saudações @mateusmachadoinvestimentos estou tentando criar uma API em java mas preciso de algumas informações podes me ajudar com isso ?
a2xevolution@hotmail.com
